### PR TITLE
Default min crew - fix for issue #232

### PR DESCRIPTION
**Pull request by ACReeser**
Instead of setting IsCommandStation to false when the RTCommandMinCrew errors out, default to 6 and note it in the log.

### CODE REVIEW

**Comment by Peppie23** [comment]
Reference #232

**Comment by Peppie23** [comment]
hm i can't get the exception notify. How can i reproduce #232 without using your vessel? (it's no stock vessel)
This is my base:
screenshot25

screenshot26

**Comment by ACReeser** [comment]
Unfortunately I have no idea how to reproduce it with stock.

I'm running a lot of mods, and using --force-opengl as well. I honestly have no idea *why* the exception is being triggered, I just think this is a better on-error behavior (in my humble opinion) because:

* Current exception behavior fails when looking up the minimum crew count, a property which in stock RemoteTech is always 6 (even though the idea is to open this up to other parts, right?). This reduces the 'my command station is broken' failure from all RT users to RT + other mod users where this other mod has a probe that requires less than 6 kerbals. Admittedly, it could also cause 'my station can now be a command station when it normally can't' if there is some part out there with a minimum crew count of greater than 6.
* Personally, when choosing between 'better for some' and 'broken for all' I'd go with the former (and this is entirely my opinion).
* A command station is a big in-game investment, and having it go kaput can be mission-ruining.

tl;dr In my opinion this is a net positive change, even if it doesn't fix the problem outright. Also, I realize this is basically down to preference, and I won't be offended if this isn't accepted. If it makes this issue more visible, then I'm happy.

**Comment by ACReeser** [comment]
My mod list if that helps:

* ActiveTextureManagement
* B9_Aerospace
* BahaSP
* Chatterer
* CommunityResourcePack
* CrossFeedEnabler
* DMagic Orbital Science
* EnhancedNavBall
* Firespitter
* JSI
* KAS
* KerbalEngineer
* KineTechAnimation
* Klockheed_Martian_Gimbal
* MP_Nazari
* NEAR
* NearFutureElectrical
* NearFutureSolar
* ProceduralFairings
* R&SCapsuledyne
* RealChute
* Regolith
* RemoteTech
* ResGen
* RLA_Stockalike
* SCANsat
* SmokeScreen
* StationPartsExpansion
* ThunderAerospace
* TriggerTech
* UniversalStorage
* Virgin Kalactic

File: src/RemoteTech/Modules/ProtoSignalProcessor.cs [hunk 1]
```diff
@@ -25,16 +26,18 @@ public ProtoSignalProcessor(ProtoPartModuleSnapshot ppms, Vessel v)
...
-                IsCommandStation = Powered && v.HasCommandStation() && v.GetVesselCrew().Count >= ppms.GetInt("RTCommandMinCrew");
+                commandMinCrew = ppms.GetInt("RTCommandMinCrew");
                 RTLog.Notify("ProtoSignalProcessor(Powered: {0}, HasCommandStation: {1}, Crew: {2}/{3})", 
-                    Powered, v.HasCommandStation(), v.GetVesselCrew().Count, ppms.GetInt("RTCommandMinCrew"));
+                    Powered, v.HasCommandStation(), v.GetVesselCrew().Count, commandMinCrew));
             } catch (ArgumentException) {
                 // I'm assuming this would get thrown by ppms.GetInt()... do the other functions have an exception spec?
-                IsCommandStation = false;
-                RTLog.Notify("ProtoSignalProcessor(Powered: {0}, HasCommandStation: {1}, Crew: {2})", 
+                RTLog.Notify("ProtoSignalProcessor(Powered: {0}, HasCommandStation: {1}, Crew: {2}/6 [Default]",
```
Comment (erendrake) [inline src/RemoteTech/Modules/ProtoSignalProcessor.cs]: you have {2}/6 hard coded in here. you should at least reference your constant from above otherwise if that changes we wont see it in the message. but really it should be reading that value out of the cfg.

Comment (ACReeser) [inline src/RemoteTech/Modules/ProtoSignalProcessor.cs]: Oh, good catch. 

I don't really know the API or the code, by 'out of the cfg' do you meansomething other than ppms.GetInt() ?

**Comment by Peppie23** [comment]
oh yea lots of mods :cold_sweat:
Anyway, can you please set your hard coded `6` into the RTSettings class? So we have the possibility to change this value using the settings.cfg file.

**Comment by Peppie23** [comment]
And if one mod deletes our value `RTCommandMinCrew` on the large probe core, maybe with module manager? Would that be possible?

**Comment by ACReeser** [comment]
Sure I'll put it in the RTSettings class. Should I be able to check ```RTCommandMinCrew``` by looking at a save file with the broken vessel?

**Comment by Peppie23** [comment]
yes please take a look at your save, can you also *search in files* for `probeStackLarge` over your GameData Directory? I would like to know what mod will change this probe core. :+1:

**Comment by jdmj** [comment]
Weird. I thought I replied to this PR through email. Anyway...
I thought the issue has been narrowed down in the RT forum thread. Bases lose their command abilities when unloaded if the probe core isn't a root part of the vessel (which also breaks things on docking/undocking). Someone even tested this idea, editing the save file, replacing the craft which had a problem with the same craft but with probe core as root, and it worked. Until the first docking/undocking, that is.

I don't think _any_ mod adds probe cores with command station ability or changes RT config for the RC-L01, which is the only stock command-capable core.

**Comment by ACReeser** [comment]
OK I ctrl+shift+f'd through GameData for ```probeStackLarge``` and I don't see any other .cfgs except for RemoteTech's.  

I verified that my save file has ```RTCommandMinCrew``` named correctly and set to 6.

@jdmj, does this mean any docking/undocking will mess cause the bug? Or am I misreading your comment? I'll have to go through the whole thread to get some context.

I'll be adding that default to the settings file shortly.

**Comment by jdmj** [comment]
> does this mean any docking/undocking will mess cause the bug?

Found where it was: http://forum.kerbalspaceprogram.com/threads/83305-0-25-0-RemoteTech-v1-5-1?p=1550153&viewfull=1#post1550153
A couple of replies later it's mentioned that docking/undocking breaks things again.

**Comment by Peppie23** [comment]
Changed destination from master to 1.6 (#276)
thx @ACReeser for your pr

**Comment by ACReeser** [comment]
Heyo! Thanks everyone for the help!
